front/src/applications/operationalStudies/hooks/toggleSelection.ts

```diff
@@ -1,17 +1,40 @@
-import type React from 'react';
+import { useCallback, useState } from 'react';
 
-const toggleSelection = (
-  id: number | undefined,
-  selectedIds: number[],
-  setSelectedIds: React.Dispatch<React.SetStateAction<number[]>>
+import type {
+  ProjectWithStudies,
+  ScenarioWithCountTrains,
+  SearchResultItemProject,
+  StudyWithScenarios,
+} from 'common/api/osrdEditoastApi';
+
+const useMultiSelection = <
+  T extends
+    | ProjectWithStudies
+    | SearchResultItemProject
+    | ScenarioWithCountTrains
+    | StudyWithScenarios,
```

Comment:
May be we should make T extends `{id: number} `to make more generic instead of only type of `Projet`, `Study` or `Scenario` ?